tools/psa/release.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/python
+# Copyright (c) 2017-2018 ARM Limited
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import subprocess
+import sys
+import shutil
+
+ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir, os.pardir))
+sys.path.insert(0, ROOT)
+from tools.targets import Target, TARGET_MAP, TARGET_NAMES
+
+PYTHON_EXEC = 'python' if os.path.basename(sys.executable).startswith('mbed') else sys.executable
+MAKE_PY_LOCATTION = os.path.join(ROOT, 'tools', 'make.py')
+TEST_PY_LOCATTION = os.path.join(ROOT, 'tools', 'test.py')
+MBED_PSA_TESTS = '*psa-spm*,*psa-crypto_access_control'
+TFM_MBED_APP = os.path.join(ROOT, 'tools', 'psa', 'tfm', 'mbed_app.json')
+TFM_TESTS = {
+    '*psa-spm_smoke': ['USE_PSA_TEST_PARTITIONS', 'USE_SMOKE_TESTS_PART1'],
+    '*psa-spm_client': ['USE_PSA_TEST_PARTITIONS', 'USE_CLIENT_TESTS_PART1'],
+    '*psa-spm_server': ['USE_PSA_TEST_PARTITIONS', 'USE_SERVER_TESTS_PART1', 'USE_SERVER_TESTS_PART2'],
+    '*psa-crypto_access_control': ['USE_PSA_TEST_PARTITIONS', 'USE_CRYPTO_ACL_TEST']
+}
+
+
+def get_mbed_official_psa_release():
+    def psa_backend(target_name):
+        return 'TFM' if 'TFM' in Target.get_target(target_name).labels else 'MBED_SPM'
+
+    psa_targets_release_dict = {
+        'TFM': [],
+        'MBED_SPM': []
+    }
+
+    psa_secure_targets = [t for t in TARGET_NAMES if Target.get_target(t).is_PSA_secure_target]
+    for t in psa_secure_targets:
+        psa_targets_release_dict[psa_backend(t)].append(
+            tuple(
+                [
+                    TARGET_MAP[t].name,
+                    TARGET_MAP[t].default_toolchain
+                ]
+            )
+        )
+
+    return psa_targets_release_dict
+
+
+def create_mbed_ignore(dir):
+    with open(os.path.join(dir, '.mbedignore'), 'w') as f:
+        f.write('*\n')
+
+
+def build_mbed_spm_platform(target, toolchain):
+    subprocess.call([
+        PYTHON_EXEC,
+        '-u',
```

Comment:
Is there a reason you chose to call `test.py` instead of invoke the python directly? Both are within the same directory after all.

Reply:
good catch missed that

Reply:
I'm not sure I was clear here: I mean you can import the functions that `test.py` calls and call them as well.

Reply:
main() function?

Reply:
I was thinking of `build_project` and `build_tests` and `build_library`.

Reply:
and then i'll need to update this tool everytime the internals change ( true, that's rare to happen but better safe than sorry)